pkg/metrics/delivery_reporter.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics"
+)
+
+type DeliveryReporter struct {
+	podName               PodName
+	containerName         ContainerName
+	dispatchTimeInMsecM   *stats.Float64Measure
+	processingTimeInMsecM *stats.Float64Measure
+}
+
+type DeliveryReportArgs struct {
+	Namespace  string
+	Broker     string
+	Trigger    string
+	FilterType string
+}
+
+func (r *DeliveryReporter) register() error {
+	return view.Register(
+		&view.View{
+			Name:        "event_count",
+			Description: "Number of events delivered to a Trigger subscriber",
+			Measure:     r.dispatchTimeInMsecM,
+			Aggregation: view.Count(),
+			TagKeys: []tag.Key{
+				NamespaceNameKey,
+				BrokerNameKey,
+				TriggerNameKey,
+				TriggerFilterTypeKey,
+				ResponseCodeKey,
+				ResponseCodeClassKey,
+				PodNameKey,
+				ContainerNameKey,
+			},
+		},
+		&view.View{
+			Description: r.dispatchTimeInMsecM.Description(),
```

Comment:
nit: The view will by default inherit Name and Description from the Measure, let's either explicitly add `Name:r.dispatchTimeInMsecM.Name()` or remove the Description. I personally prefer explicitly add them here.

Reply:
Added the names explicitly